test/circuit/v2/hop.spec.ts

```diff
@@ -0,0 +1,741 @@
+      expect(data).to.have.length(1)
+      expect(data[0]).to.have.length(5)
+      expect(srcServerAbort.callCount).to.equal(1)
+    })
+
+    it('should connect - data limit - dest to src', async () => {
+      const hasReservationStub = sinon.stub(reservationStore, 'hasReservation')
+      const getReservationStub = sinon.stub(reservationStore, 'get')
+      hasReservationStub.resolves(true)
+      getReservationStub.resolves({
+        expire: new Date(Date.now() + 2 * 60 * 1000),
+        addr: multiaddr('/ip4/0.0.0.0'),
+        // set limit
+        limit: {
+          data: BigInt(5),
+          duration: 0
+        }
+      })
+      const dstConn = mockConnection(
+        mockMultiaddrConnection(pair<Uint8Array>(), dstPeer)
+      )
+      const [dstServer, dstClient] = duplexPair<any>()
+      const [srcServer, srcClient] = duplexPair<any>()
+
+      // resolve the destination stream for the server
+      const streamStub = sinon.stub(dstConn, 'newStream')
+      const dstServerStream = mockStream(dstServer)
+      const dstServerStreamAbortStub = sinon.spy(dstServerStream, 'abort')
+      streamStub.resolves(dstServerStream)
+
+      const stub = sinon.stub(components.connectionManager, 'getConnections')
+      stub.returns([dstConn])
+      const handleHop = expect(handleHopProtocol({
+        connection: conn,
+        stream: pbStream(mockStream(srcServer)),
+        request: {
+          type: HopMessage.Type.CONNECT,
+          peer: {
+            id: dstPeer.toBytes(),
+            addrs: []
+          }
+        },
+        relayPeer: relayPeer,
+        relayAddrs: [],
+        reservationStore,
+        peerStore: components.peerStore,
+        connectionManager: components.connectionManager
+      })).to.eventually.fulfilled()
+
+      const dstClientPbStream = pbStream(dstClient)
+      const stopConnectRequest = await dstClientPbStream.pb(StopMessage).read()
+      expect(stopConnectRequest.type).to.eq(StopMessage.Type.CONNECT)
+      // write response
+      dstClientPbStream.pb(StopMessage).write({
+        type: StopMessage.Type.STATUS,
+        status: Status.OK
+      })
+
+      await handleHop
+      const srcClientPbStream = pbStream(srcClient)
+      const response = await srcClientPbStream.pb(HopMessage).read()
+      expect(response.type).to.be.equal(HopMessage.Type.STATUS)
+      expect(response.status).to.be.equal(Status.OK)
+
+      const sourceStream = srcClientPbStream.unwrap()
+      const destStream = dstClientPbStream.unwrap()
+
+      // dest to source, write 4 bytes
+      const sender = pushable()
+      void pipe(sender, destStream)
+      sender.push(uint8arrayFromString('01234'))
+      // dest to source, exceed stream limit
+      sender.push(uint8arrayFromString('extra'))
+      const data = await all(sourceStream.source)
+      expect(data).to.have.length(1)
+      expect(data[0]).to.have.length(5)
+      expect(dstServerStreamAbortStub.callCount).to.equal(1)
```

Comment:
Same comment as elsewhere, this sends five bytes not four and the reset needs to be asserted on in both directions.